### PR TITLE
Added Automatic Documentation Generation

### PR DESCRIPTION
**Pull request by quantum9Innovation**
# DOCS
Part of the epispot CI Overhaul
___

## New Feature
On push to the `nightly` branch, automatic documentation will be generated via `pdoc3`, stashed, and then pushed to the `gh-pages` branch from where it will be deployed. This also makes it easier to set up localization services such as GitLocalized.
In order to make documentation better, docstrings were changed to add more spacing and use parameter lists (see DOCUMENTATION.md).
## Known Issues
No known issues.
## Code Breakdown
 - .github/workflows/docs.yml
    - Generates automatic documentation and pushes to `gh-pages`
 - .gitignore
    - Ignore `docs/` folder so documentation previews are not published to `nightly`
 - DOCUMENTATION.md
    - Added documentation standards
 - epispot/*
    - __init__
        - Changed version to `v2.1.0.0`
        - Changed `__stable__` to `False`
        - Added docstrings to explain each of the setup variables
    - Cleaned up package documentation
 - 
## Additional Notes
List any additional notes here.
You can also use this section to **@mention** a CODEOWNER or link to an issue.
If this PR *solves* an issue, use closing keywords here. Ex:
**Fixes #x**

### CODE REVIEW

File: .gitignore
Comment (Quantalabs) [inline .gitignore:26]: Why do you have `docs/` and `docs/*`? `docs/` already ignores the whole folder!

File: epispot/__init__.py
Comment (Quantalabs) [inline epispot/__init__.py:28]: Why do you have a docstring for this?

File: epispot/models.py
Comment (Quantalabs) [inline epispot/models.py:43]: If you put `[]` in \`\` then why don't you do this everywhere else? Such as on lines 32 and 31.

**Comment by quantum9Innovation** [comment]
Ignoring CodeFactor (as part of CI Overahaul CodeQL providers are likely to switch)

File: epispot/models.py
Comment (Quantalabs) [inline epispot/models.py:61]: You should change all of the `:return:` directives to `- returns:` instead of `- return:`.

**Comment by quantum9Innovation** [comment]
Waiting on a review and LGTM

File: epispot/pre.py
Comment (Quantalabs) [inline epispot/pre.py:6]: Bold the `STRUCTURE` label, or make it a heading. **`STRUCTURE`** or
 ### Structure

**Review by Quantalabs — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (quantum9Innovation) [reply .gitignore:26]: Ok sounds good, will change to `docs/`

File: epispot/__init__.py
Comment (quantum9Innovation) [reply epispot/__init__.py:28]: So it displays on documentation

File: epispot/models.py
Comment (quantum9Innovation) [reply epispot/models.py:43]: Will add in future PR, for now we just need basic docs

File: epispot/models.py
Comment (quantum9Innovation) [reply epispot/models.py:61]: Unnecessary

File: epispot/__init__.py
Comment (Quantalabs) [reply epispot/__init__.py:28]: It's unnecassary

Comment (quantum9Innovation) [reply epispot/__init__.py:28]: It is necessary because it shows how to access the GitHub source URL from the package itself.

**Review by Quantalabs — APPROVED** [review]

File: epispot/pre.py
Comment (quantum9Innovation) [reply epispot/pre.py]: Done! :heavy_check_mark:

**Pull request merged**
